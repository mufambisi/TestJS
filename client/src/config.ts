export enum Toggles {
  spikeWidget = "SpikeWidget",
}

/* eslint-disable no-undef */
export const config: IAppConfig =
  typeof TAL === "undefined"
    ? typeof (window as any).__STORYBOOK_ADDONS === "undefined"
      ? (null as any)
      : {}
    : TAL.config;

if (config == null) {
  throw new Error("App config is missing");
}
export const appName = TAL.config.appName || "dashboard-rx";
