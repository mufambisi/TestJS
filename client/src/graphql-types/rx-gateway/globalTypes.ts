/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum DigitalAccountStatus {
  ACTIVE = "ACTIVE",
  CANCELLED = "CANCELLED",
  CLOSED = "CLOSED",
  SETUP = "SETUP",
  UNKNOWN = "UNKNOWN",
}

/**
 * Represents a type of origin system this account belongs to.
 */
export enum DigitalAccountType {
  BROADBAND = "BROADBAND",
  KRAKEN = "KRAKEN",
  LPG = "LPG",
  OHM = "OHM",
  SAP = "SAP",
  UNKNOWN = "UNKNOWN",
  VPP = "VPP",
}

export enum DigitalServiceStatus {
  ACTIVE = "ACTIVE",
  CANCELLED = "CANCELLED",
  CLOSED = "CLOSED",
  SETUP = "SETUP",
  UNKNOWN = "UNKNOWN",
}

export enum DigitalServiceType {
  ELECTRICITY = "ELECTRICITY",
  GAS = "GAS",
  HOT_WATER = "HOT_WATER",
  LPG_EXCHANGE = "LPG_EXCHANGE",
  LPG_EXCHANGE_METERED = "LPG_EXCHANGE_METERED",
  LPG_INSITU = "LPG_INSITU",
  LPG_INSITU_METERED = "LPG_INSITU_METERED",
  LPG_METERED = "LPG_METERED",
  LPG_TERMINAL = "LPG_TERMINAL",
  NBN = "NBN",
  OHM = "OHM",
  OPTICOMM = "OPTICOMM",
  SOLAR = "SOLAR",
  UNKNOWN = "UNKNOWN",
  VOIP = "VOIP",
  VPP_BATTERY = "VPP_BATTERY",
}

export enum DigitalUserCustomerType {
  MULTI_SITE_PORTAL = "MULTI_SITE_PORTAL",
  RESIDENTIAL = "RESIDENTIAL",
  SMALL_MEDIUM_ENTERPRISE = "SMALL_MEDIUM_ENTERPRISE",
}

export enum FeatureStatus {
  ACTIVE = "ACTIVE",
  INACTIVE = "INACTIVE",
}

export enum FeatureType {
  BATTERY = "BATTERY",
  EV = "EV",
  INTERCOM = "INTERCOM",
  OHM = "OHM",
}

export enum ServiceType {
  ELECTRICITY = "ELECTRICITY",
  GAS = "GAS",
  HOT_WATER = "HOT_WATER",
}

//==============================================================
// END Enums and Input Objects
//==============================================================
