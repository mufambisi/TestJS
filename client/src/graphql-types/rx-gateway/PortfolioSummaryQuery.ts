/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ServiceType } from "./globalTypes";

// ====================================================
// GraphQL query operation: PortfolioSummaryQuery
// ====================================================

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService_property_address_components {
  __typename: "AustralianAddressComponents";
  postcode: string | null;
  locality: string | null;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService_property_address {
  __typename: "AustralianAddress";
  value: string;
  components: PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService_property_address_components;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService_property {
  __typename: "Property";
  address: PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService_property_address;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService {
  __typename: "ElectricityService";
  type: ServiceType;
  validFrom: any | null;
  isActive: boolean;
  property: PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService_property | null;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService_property_address_components {
  __typename: "AustralianAddressComponents";
  postcode: string | null;
  locality: string | null;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService_property_address {
  __typename: "AustralianAddress";
  value: string;
  components: PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService_property_address_components;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService_property {
  __typename: "Property";
  address: PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService_property_address;
}

export interface PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService {
  __typename: "GasService";
  type: ServiceType;
  validFrom: any | null;
  isActive: boolean;
  property: PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService_property | null;
}

export type PortfolioSummaryQuery_viewer_kraken_accounts_services = PortfolioSummaryQuery_viewer_kraken_accounts_services_ElectricityService | PortfolioSummaryQuery_viewer_kraken_accounts_services_GasService;

export interface PortfolioSummaryQuery_viewer_kraken_accounts {
  __typename: "Account";
  accountNumber: string;
  services: PortfolioSummaryQuery_viewer_kraken_accounts_services[];
}

export interface PortfolioSummaryQuery_viewer_kraken {
  __typename: "KrakenViewer";
  accounts: PortfolioSummaryQuery_viewer_kraken_accounts[];
}

export interface PortfolioSummaryQuery_viewer {
  __typename: "Viewer";
  kraken: PortfolioSummaryQuery_viewer_kraken | null;
}

export interface PortfolioSummaryQuery {
  viewer: PortfolioSummaryQuery_viewer;
}
