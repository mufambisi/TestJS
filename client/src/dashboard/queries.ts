import { gql, useKraxiQuery } from "@origin-digital/kraxi";
import { PortfolioSummaryQuery } from "../graphql-types/rx-gateway/PortfolioSummaryQuery";
import { UserAccounts } from "../graphql-types/rx-gateway/UserAccounts";

export const useUserAccountsQuery = () =>
  useKraxiQuery<UserAccounts>(gql`
    query UserAccounts {
      viewer {
        digital {
          services {
            status
            type
          }
          accounts {
            id
            type
            accountId
            status
          }
          features {
            id
            type
            featureId
            status
          }
          user {
            customerType
          }
        }
      }
    }
  `);

export const usePortfolioSummaryQuery = () =>
  useKraxiQuery<PortfolioSummaryQuery>(
    gql`
      query PortfolioSummaryQuery {
        viewer {
          kraken {
            accounts {
              accountNumber
              services {
                ... on ElectricityService {
                  type
                  validFrom
                  isActive
                  property {
                    address {
                      value
                      components {
                        postcode
                        locality
                      }
                    }
                  }
                }
                ... on GasService {
                  type
                  validFrom
                  isActive
                  property {
                    address {
                      value
                      components {
                        postcode
                        locality
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    `,
    {
      // bypassing cache to avoid useUserAccountsQuery from excuting again
      fetchPolicy: "no-cache",
    }
  );
