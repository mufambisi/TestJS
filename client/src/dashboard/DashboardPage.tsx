import React, { useEffect, useRef, useState } from "react";

import { CardStackSection } from "@origin-digital/ods-core";
import { useToggleContext } from "@origin-digital/toggle";

import { Loader } from "../components/Loader";
import { LpgFeatureMissingBanner } from "../components/LpgFeatureMissingBanner";
import { DigitalAccountType } from "../graphql-types/rx-gateway/globalTypes";
import { useUserAccountsQuery } from "./queries";
import { DashboardMonitoring } from "./widgetHelpers/widgetAnalytics";
import { getWidgetManifest, WidgetManifest } from "./widgetManifest";
import { WidgetLoader } from "./WidgetLoader";

export const DashboardPage: React.FC = () => {
  const { data, loading, error } = useUserAccountsQuery();
  const [widgetManifest, setWidgetManifest] = useState<WidgetManifest>();
  const dashboardAnalytics = useRef<DashboardMonitoring>();
  const { toggles } = useToggleContext();

  useEffect(() => {
    if (!data) {
      return;
    }
    const filteredWidgetManifests = getWidgetManifest(
      toggles,
      data.viewer.digital
    );

    dashboardAnalytics.current = new DashboardMonitoring(
      data,
      filteredWidgetManifests
    );
    setWidgetManifest(filteredWidgetManifests);
  }, [data]);

  if (error) {
    throw error;
  }

  const digitalData = data?.viewer.digital;
  const hasLpg = digitalData?.accounts?.some(
    (account) => account.type === DigitalAccountType.LPG
  );

  return (
    <Loader isLoading={loading} variant="page">
      {hasLpg ? <LpgFeatureMissingBanner /> : null}
      {widgetManifest && digitalData ? (
        <CardStackSection data-id="dashboard" variant="widget">
          {widgetManifest.map((widgetDefinition) => (
            <WidgetLoader
              key={widgetDefinition.name}
              dashboardAnalytics={dashboardAnalytics}
              widgetDefinition={widgetDefinition}
              userData={digitalData}
            />
          ))}
        </CardStackSection>
      ) : null}
    </Loader>
  );
};
