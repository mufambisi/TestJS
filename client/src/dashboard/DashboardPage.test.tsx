import { render } from "@origin-digital/ods-testing-library";
import React from "react";
import { ServiceType } from "src/graphql-types/rx-gateway/globalTypes";
import { DashboardPage } from "./DashboardPage";
import { usePortfolioSummaryQuery, useUserAccountsQuery } from "./queries";
import { config } from "../config";
import { ToggleContextProvider } from "@origin-digital/toggle";

jest.mock("@origin-digital/event-dispatcher", () => ({
  ...jest.requireActual("@origin-digital/event-dispatcher"),
  eventListener: { addListener: jest.fn() },
}));
jest.mock("@origin-digital/kraken-account", () => ({
  ...jest.requireActual("@origin-digital/kraken-account"),
  AccountCard: ({ accountNumber }: { accountNumber: string }) => (
    <div data-id="service-id">{accountNumber}</div>
  ),
}));
jest.mock("src/util/analytics");
jest.mock("src/util/jwt");
jest.mock("./queries");

const accountNumber = "A-132645879";

const renderDashboardPage = () =>
  render(
    <ToggleContextProvider toggles={config.toggles} appName={config.appName}>
      <DashboardPage />
    </ToggleContextProvider>
  );

describe("<DashboardPage />", () => {
  beforeEach(() => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      data: {
        viewer: {
          kraken: {
            accounts: [
              {
                accountNumber,
                services: [{ __typename: "ElectricityService" }],
              },
            ],
          },
          digital: {
            accounts: [
              {
                id: "1",
                type: "KRAKEN",
                accountId: accountNumber,
                product: "ELECTRICITY",
                status: "ACTIVE",
              },
            ],
            features: [],
            services: [],
          },
        },
      },
    });

    (usePortfolioSummaryQuery as jest.Mock).mockReturnValue({
      data: {
        viewer: {
          kraken: {
            accounts: [
              {
                __typename: "Account",
                accountNumber,
                services: [
                  {
                    __typename: "ElectricityService",
                    type: ServiceType.ELECTRICITY,
                    validFrom: "2020-12-31T14:00:00.000Z",
                    isActive: true,
                    property: {
                      __typename: "Property",
                      address: {
                        __typename: "AustralianAddress",
                        value: "U 3, 76 Mathoura Rd, Toorak, VIC 3142",
                        components: {
                          __typename: "AustralianAddressComponents",
                          locality: "Toorak",
                          postcode: "3142",
                        },
                      },
                    },
                  },
                ],
              },
            ],
          },
        },
      },
    });
  });

  it("should render without error", () => {
    const { container } = renderDashboardPage();

    expect(container).toMatchSnapshot();
  });

  it("should contain correct account number", () => {
    const { getByTestId } = renderDashboardPage();

    expect(getByTestId("service-id").textContent).toBe("A-132645879");
  });

  it("should render error message", () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      error: {},
    });

    expect(() => renderDashboardPage()).toThrowError();
  });

  it("should render the loading indicator when loading", () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      loading: true,
    });

    const { getAllByTestId } = renderDashboardPage();

    expect(getAllByTestId("page-loading-indicator")).toHaveLength(1);
  });

  it("should not render the loading indicator when loading is false", () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      loading: false,
    });

    const { queryAllByTestId } = renderDashboardPage();

    expect(queryAllByTestId("page-loading-indicator")).toHaveLength(0);
  });

  it("should render LPG missing feature banner and not default missing feature banner", async () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      data: {
        viewer: {
          kraken: {
            accounts: [
              {
                accountNumber,
                services: [{ __typename: "ElectricityService" }],
              },
            ],
          },
          digital: {
            accounts: [
              {
                type: "KRAKEN",
              },
              {
                type: "LPG",
              },
            ],
            features: [],
            services: [],
          },
        },
      },
    });
    const { findByTestId } = renderDashboardPage();

    await findByTestId("lpg-missing-features-banner");
  });

  it("should render split crm card when there is at least one active sap, and one active kraken acccount", async () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      data: {
        viewer: {
          kraken: {
            accounts: [
              {
                accountNumber: "A-5CA6C8DE",
                services: [
                  {
                    __typename: "GasService",
                    type: "GAS",
                    validFrom: "2020-12-31T14:00:00.000Z",
                    isActive: true,
                    property: {
                      address: {
                        value: "24 West St, Hillsdale, Nsw 2047",
                        components: {
                          locality: "Hillsdale",
                          postcode: "2047",
                        },
                      },
                    },
                  },
                ],
              },
            ],
          },
          digital: {
            accounts: [
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjExZWJmNWE5LTVmZjEtMDJjOS04N2M0LTA2NzM0YjA1MDgzKjM9",
                type: "SAP",
                accountId: "400005194727",
                product: "ELECTRICITY",
                status: "ACTIVE",
              },
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjUxOTZkNGMzLWVjZGUtNDA3Mi1hMzVjLWIxYzY4M2E3OGExOCJ1",
                type: "KRAKEN",
                accountId: "A-5CA6C8DE",
                product: "GAS",
                status: "ACTIVE",
              },
            ],
            features: [],
            services: [],
          },
        },
      },
    });
    const { findByTestId } = renderDashboardPage();

    await findByTestId("split-crm-card");
  });

  it("should render split crm card with no bullet points, when there is only one active sap and one active kraken account", async () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      data: {
        viewer: {
          kraken: {
            accounts: [
              {
                accountNumber: "A-5CA6C8DE",
                services: [
                  {
                    __typename: "GasService",
                    type: "GAS",
                    validFrom: "2020-12-31T14:00:00.000Z",
                    isActive: true,
                    property: {
                      address: {
                        value: "24 West St, Hillsdale, Nsw 2047",
                        components: {
                          locality: "Hillsdale",
                          postcode: "2047",
                        },
                      },
                    },
                  },
                ],
              },
            ],
          },
          digital: {
            accounts: [
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjExZWJmNWE5LTVmZjEtMDJjOS04N2M0LTA2NzM0YjA1MDgzKjM9",
                type: "SAP",
                accountId: "400005194727",
                product: "ELECTRICITY",
                status: "ACTIVE",
              },
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjUxOTZkNGMzLWVjZGUtNDA3Mi1hMzVjLWIxYzY4M2E3OGExOCJ1",
                type: "KRAKEN",
                accountId: "A-5CA6C8DE",
                product: "GAS",
                status: "ACTIVE",
              },
            ],
            features: [],
            services: [],
          },
        },
      },
    });
    const { findByTestId, queryByRole } = renderDashboardPage();

    await findByTestId("split-crm-card");
    const dotPoints = queryByRole("listitem");
    expect(dotPoints).not.toBeInTheDocument();
  });

  it("should render split crm card with bullet points, when there is more than one active sap and one active kraken account", async () => {
    (useUserAccountsQuery as jest.Mock).mockReturnValue({
      data: {
        viewer: {
          kraken: {
            accounts: [
              {
                accountNumber: "A-5CA6C8DE",
                services: [
                  {
                    __typename: "GasService",
                    type: "GAS",
                    validFrom: "2020-12-31T14:00:00.000Z",
                    isActive: true,
                    property: {
                      address: {
                        value: "24 West St, Hillsdale, Nsw 2047",
                        components: {
                          locality: "Hillsdale",
                          postcode: "2047",
                        },
                      },
                    },
                  },
                ],
              },
            ],
          },
          digital: {
            accounts: [
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjExZWJmNWE5LTVmZjEtMDJjOS04N2M0LTA2NzM0YjA1MDgzKjM9",
                type: "SAP",
                accountId: "400005194727",
                product: "ELECTRICITY",
                status: "ACTIVE",
              },
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjExZWJmNWE5LTVmZjEtMDJjOS04N2M0LTA2NzM0YjA1MDgzKjM8",
                type: "SAP",
                accountId: "400005194728",
                product: "ELECTRICITY",
                status: "ACTIVE",
              },
              {
                id:
                  "eyJ0eXBlIjoiRGlnaXRhbEFjY291bnQiLCJpZCI6IjUxOTZkNGMzLWVjZGUtNDA3Mi1hMzVjLWIxYzY4M2E3OGExOCJ1",
                type: "KRAKEN",
                accountId: "A-5CA6C8DE",
                product: "GAS",
                status: "ACTIVE",
              },
            ],
            features: [],
            services: [],
          },
        },
      },
    });
    const { findByTestId, queryAllByRole } = renderDashboardPage();

    await findByTestId("split-crm-card");
    const dotPoints = queryAllByRole("listitem");
    expect(dotPoints).toHaveLength(2);
  });
});
