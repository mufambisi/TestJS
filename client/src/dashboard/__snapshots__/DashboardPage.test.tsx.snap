// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardPage /> should render without error 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  width: 100%;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #fa4616;
  width: 100%;
}

.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: -88px;
  margin-bottom: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c6 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
  width: 100%;
}

.c8 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c9 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

.c12 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c16 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: transparent;
  padding-top: 8px;
  padding-bottom: 8px;
  width: 100%;
}

.c15 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c13 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c14 {
  max-width: 100%;
}

.c7:empty {
  display: none;
}

.c7:empty + div:before {
  display: none;
}

.c7:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  margin-top: 16px;
}

.c7:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  margin-top: 16px;
}

.c10:empty {
  display: none;
}

.c10:empty + div:before {
  display: none;
}

.c10:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  margin-top: 16px;
}

.c10:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  margin-top: 16px;
}

.c11 {
  width: 100%;
  min-width: 10px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:992px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:992px) {
  .c3 {
    margin-top: 48px;
  }
}

@media screen and (min-width:992px) {
  .c3 {
    margin-bottom: 48px;
  }
}

@media screen and (min-width:0px) {
  .c9 {
    border-radius: 8px;
  }
}

@media screen and (min-width:576px) {
  .c9 {
    border-radius: 8px;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    border-radius: 8px;
  }
}

@media screen and (min-width:992px) {
  .c9 {
    border-radius: 8px;
  }
}

@media screen and (min-width:1200px) {
  .c9 {
    border-radius: 8px;
  }
}

@media screen and (min-width:992px) {
  .c9 {
    padding-top: 32px;
  }
}

@media screen and (min-width:992px) {
  .c9 {
    padding-bottom: 32px;
  }
}

@media screen and (min-width:576px) {
  .c9 {
    padding-left: 32px;
  }
}

@media screen and (min-width:576px) {
  .c9 {
    padding-right: 32px;
  }
}

@media screen and (min-width:576px) {
  .c16 {
    padding-top: 8px;
  }
}

@media screen and (min-width:768px) {
  .c16 {
    padding-top: 8px;
  }
}

@media screen and (min-width:992px) {
  .c16 {
    padding-top: 0;
  }
}

@media screen and (min-width:1200px) {
  .c16 {
    padding-top: 0;
  }
}

@media screen and (min-width:576px) {
  .c16 {
    padding-bottom: 8px;
  }
}

@media screen and (min-width:768px) {
  .c16 {
    padding-bottom: 8px;
  }
}

@media screen and (min-width:992px) {
  .c16 {
    padding-bottom: 0;
  }
}

@media screen and (min-width:1200px) {
  .c16 {
    padding-bottom: 0;
  }
}

@media screen and (min-width:576px) {
  .c7:not(:empty) ~ div:not(:empty):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:768px) {
  .c7:not(:empty) ~ div:not(:empty):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:992px) {
  .c7:not(:empty) ~ div:not(:empty):before {
    margin-top: 32px;
  }
}

@media screen and (min-width:1200px) {
  .c7:not(:empty) ~ div:not(:empty):before {
    margin-top: 32px;
  }
}

@media screen and (min-width:576px) {
  .c7:nth-child(n + 2):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:768px) {
  .c7:nth-child(n + 2):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:992px) {
  .c7:nth-child(n + 2):before {
    margin-top: 32px;
  }
}

@media screen and (min-width:1200px) {
  .c7:nth-child(n + 2):before {
    margin-top: 32px;
  }
}

@media screen and (min-width:576px) {
  .c10:not(:empty) ~ div:not(:empty):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:768px) {
  .c10:not(:empty) ~ div:not(:empty):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:992px) {
  .c10:not(:empty) ~ div:not(:empty):before {
    margin-top: 24px;
  }
}

@media screen and (min-width:1200px) {
  .c10:not(:empty) ~ div:not(:empty):before {
    margin-top: 24px;
  }
}

@media screen and (min-width:576px) {
  .c10:nth-child(n + 2):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:768px) {
  .c10:nth-child(n + 2):before {
    margin-top: 16px;
  }
}

@media screen and (min-width:992px) {
  .c10:nth-child(n + 2):before {
    margin-top: 24px;
  }
}

@media screen and (min-width:1200px) {
  .c10:nth-child(n + 2):before {
    margin-top: 24px;
  }
}

@media (min-width:576px) {
  .c4 {
    max-width: 508px;
  }
}

@media (min-width:768px) {
  .c4 {
    max-width: 448px;
  }
}

@media (min-width:992px) {
  .c4 {
    max-width: 448px;
  }
}

@media (min-width:1200px) {
  .c4 {
    max-width: 538px;
  }
}

@media screen and (min-width:0px) {
  .c2 {
    margin-top: -88px;
  }
}

@media screen and (min-width:992px) {
  .c2 {
    margin-top: 48px;
  }
}

@media screen and (min-width:0px) {
  .c2 {
    margin-bottom: 32px;
  }
}

@media screen and (min-width:992px) {
  .c2 {
    margin-bottom: 48px;
  }
}

<div>
  <div
    class="c0"
    data-id="dashboard"
    width="full"
  >
    <div
      class="c1"
      display=",,,none,"
      style="height: 120px;"
      width="full"
    />
    <div
      class="c2 c3"
      display="flex"
      width="full"
    >
      <div
        class="c4 c5"
        display="flex"
        width="full"
      >
        <div
          class="c6"
          data-id="stack"
          display="block"
          width="full"
        >
          <div
            class="c7 c8"
            data-id="stack-child"
            display="block"
          >
            <div
              class="c6"
              data-id="stack"
              display="block"
              width="full"
            >
              <div
                class="c7 c8"
                data-id="stack-child"
                display="block"
              >
                <div
                  data-id="service-id"
                >
                  A-132645879
                </div>
              </div>
            </div>
          </div>
          <div
            class="c7 c8"
            data-id="stack-child"
            display="block"
          >
            <div
              class="c0"
              data-id="fuel-discount-widget"
              width="full"
            >
              <div
                class="c9"
                data-id="card"
                width="full"
              >
                <div
                  class="c6"
                  data-id="stack"
                  display="block"
                  width="full"
                >
                  <div
                    class="c10 c8"
                    data-id="stack-child"
                    display="block"
                  >
                    <div
                      class="c11 c12"
                      data-id="placeholder"
                      display="flex"
                      height="120"
                      overflow="hidden"
                      shape="rectangle"
                      width="100%"
                    >
                      <svg
                        style="width: 100%; height: 100%; position: absolute;"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <line
                          style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
                          x1="0"
                          x2="100%"
                          y1="0"
                          y2="100%"
                        />
                        <line
                          style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
                          x1="100%"
                          x2="0"
                          y1="0"
                          y2="100%"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="c10 c8"
                    data-id="stack-child"
                    display="block"
                  >
                    <h4
                      class="typography-h4-variant-basekick-1 typography-neutralDark-color-2 typography-strong-support-3 typography-medium-weight-4 -h4 c13"
                      data-id="heading"
                    >
                      Save 4¢ per litre at EG fuel locations
                    </h4>
                  </div>
                  <div
                    class="c10 c8"
                    data-id="stack-child"
                    display="block"
                  >
                    <span
                      class="typography-body-variant-basekick-6 typography-neutral-color-7 typography-regular-weight-8 typography-strong-support-9 c14 -span c15"
                      data-id="text"
                    >
                      Save even more by combining with other offers!
                    </span>
                  </div>
                  <div
                    class="c10 c8"
                    data-id="stack-child"
                    display="block"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root Button-root-11 Button-root-24 MuiButton-contained Button-contained-16 MuiButton-containedPrimary Button-containedPrimary-17"
                      data-id="button"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Get offer
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c7 c8"
            data-id="stack-child"
            display="block"
          />
          <div
            class="c7 c8"
            data-id="stack-child"
            display="block"
          />
          <div
            class="c7 c8"
            data-id="stack-child"
            display="block"
          >
            <div
              class="c9"
              data-id="contact-card"
              width="full"
            >
              <div
                class="c6"
                data-id="stack"
                display="block"
                width="full"
              >
                <div
                  class="c10 c8"
                  data-id="stack-child"
                  display="block"
                >
                  <h4
                    class="typography-h4-variant-basekick-39 typography-neutralDark-color-40 typography-strong-support-41 typography-medium-weight-42 -h4 c13"
                    data-id="heading"
                  >
                    Need help?
                  </h4>
                </div>
                <div
                  class="c10 c8"
                  data-id="stack-child"
                  display="block"
                >
                  <span
                    class="typography-body-variant-basekick-44 typography-neutral-color-45 typography-regular-weight-46 typography-strong-support-47 c14 -span c15"
                    data-id="text"
                  >
                    Contact us and we'll get back to you as soon as possible.
                  </span>
                </div>
                <div
                  class="c10 c8"
                  data-id="stack-child"
                  display="block"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root Button-root-11 Button-root-49 MuiButton-outlined Button-outlined-19 MuiButton-outlinedPrimary Button-outlinedPrimary-20"
                    data-id="open-contact-us"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Contact us
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c7 c8"
            data-id="stack-child"
            display="block"
          >
            <div
              class="c16"
              width="full"
            >
              <span
                class="typography-body-variant-basekick-52 typography-neutral-color-53 typography-regular-weight-54 typography-strong-support-55 c14 -span c15"
                data-id="text"
              >
                Amounts include GST although your account balance may also have non-taxable items. Check the bill we sent you for a full breakdown.
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
